authentication/http.go

```diff
@@ -93,31 +93,31 @@ func GetGroups(ctx context.Context) ([]string, bool) {
 	return groups, ok
 }
 
+// MiddlewareFn is function type able to return
+// authentication middleware for given tenant. If
```

Comment:
Nit: ```suggestion
// authentication middleware for a given tenant. If
```

Reply:
I would love if signature and naming would indicate that this returns only auth middleware. Right now it looks pretty generic

Reply:
Do you think it is not clear enough? Since when exported, it would be used as `authentication.Middleware`, making it clear this is auth middleware. Naming it e.g. `AuthMiddleware` would stutter a bit.